main/main.go

```diff
@@ -138,7 +136,9 @@ func main() {
 
 	mux := &sync.Mutex{}
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	cfg := ctrl.GetConfigOrDie()
+	cfg.Wrap(common.RetryMiddleware(log)) // allow all client calls to be retriable
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
```

Comment:
I see that ctrl.options has an option to configure a custom HTTPclient, but I think that's just another way to configure it
```
		Client: client.Options{
			HTTPClient: &http.Client{
				Transport: <your-custom-transport>,
			},
		},
```

Reply:
Yep! Also `config.Wrap()` seems to be a nice abstraction which allowed us to include the same middleware to our other clients (dynamic and clienset) pretty easily without worrying about any underlying issues/config like including the default Transporter at the end of the chain etc etc